### PR TITLE
Rpc server

### PR DESCRIPTION
**Pull request by mskrzypkows**
L2 tx lists passed by JSON RPC

added test_rpc.py to check how it works

### CODE REVIEW

**Review by smartprogrammer93 — CHANGES_REQUESTED** [review]
Looks good. One thing I would do just to confirm it works is to sync a taiko geth node with a taiko driver and then run this modified proposer next to it. Try the RPC, and see what I would get.
One other thing is, we need to disable the default functionality of taiko proposer  (proposing blocks on its own)

**Comment by mskrzypkows** [comment]
I've already tested it, here is an output for proper tx lists: https://pastebin.com/3AuUYLWr
And when we're not synce yet we receive:
```
Response: {"result":null,"error":"failed to fetch transaction pool content: execution reverted: arithmetic underflow or overflow","id":1}
```

File: packages/taiko-client/proposer/proposer.go [hunk 1]
```diff
@@ -159,11 +163,49 @@ func (p *Proposer) InitFromConfig(ctx context.Context, cfg *Config) (err error)
 
 // Start starts the proposer's main loop.
 func (p *Proposer) Start() error {
-	p.wg.Add(1)
-	go p.eventLoop()
+	startRPCServer(p)
+
+	// p.wg.Add(1)
```
Comment (mskrzypkows) [inline packages/taiko-client/proposer/proposer.go:168]: @smartprogrammer93 Here I disable default proposing, we may add some env variable for this, or just delete this code.

**Comment by mskrzypkows** [comment]
https://github.com/NethermindEth/Taiko-Preconf-AVS/issues/5

**Pull request merged**